### PR TITLE
[LTR] Implement Bewitching Leechcraft

### PR DESCRIPTION
**Pull request by Susucre**

### CODE REVIEW

**Comment by Susucre** [comment]
[[Bewitching Leechcraft]]

File: Mage.Sets/src/mage/cards/b/BewitchingLeechcraft.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+    public BewitchingLeechcraft(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{1}{U}");
+        
+        this.subtype.add(SubType.AURA);
+
+        // Enchant creature
+        TargetPermanent auraTarget = new TargetCreaturePermanent();
+        this.getSpellAbility().addTarget(auraTarget);
+        this.getSpellAbility().addEffect(new AttachEffect(Outcome.BoostCreature));
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/b/BewitchingLeechcraft.java:37]: `Outcome.Tap` would make more sense here (this is only used by the AI and may or may not make a difference)

File: Mage.Sets/src/mage/cards/b/BewitchingLeechcraft.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        this.addAbility(new EnchantAbility(auraTarget));
+
+        // When Bewitching Leechcraft enters the battlefield, tap enchanted creature.
+        this.addAbility(new EntersBattlefieldTriggeredAbility(new TapEnchantedEffect()));
+
+        // Enchanted creature has "If this creature would untap during your untap step,
+        // remove a +1/+1 counter from it instead. If you do, untap it."
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD,
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/b/BewitchingLeechcraft.java:45]: `SimpleStaticAbility` constructor without zone defaults to `Zone.BATTLEFIELD` for simplicity

File: Mage.Sets/src/mage/cards/b/BewitchingLeechcraft.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+    @Override
+    public BewitchingLeechcraft copy() {
+        return new BewitchingLeechcraft(this);
+    }
+}
+
+class BewitchingLeechcraftReplacementEffect extends ReplacementEffectImpl {
+
+    public BewitchingLeechcraftReplacementEffect() {
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/b/BewitchingLeechcraft.java:65]: can be package private

File: Mage.Sets/src/mage/cards/b/BewitchingLeechcraft.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        super(Duration.EndOfGame, Outcome.Detriment);
+        staticText = "If this creature would untap during your untap step, " +
+            "remove a +1/+1 counter from it instead. If you do, untap it.";
+    }
+
+    public BewitchingLeechcraftReplacementEffect(final BewitchingLeechcraftReplacementEffect effect) {
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/b/BewitchingLeechcraft.java:71]: can be private

File: Mage.Sets/src/mage/cards/b/BewitchingLeechcraft.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+    @Override
+    public BewitchingLeechcraftReplacementEffect copy() {
+        return new BewitchingLeechcraftReplacementEffect(this);
+    }
+
+    @Override
+    public boolean checksEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.UNTAP
+            && game.getTurnStepType() == PhaseStep.UNTAP;
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/b/BewitchingLeechcraft.java:83]: I believe `checksEventType()` should only compare `event.getType()` and all other checks should be in `applies()` method.

File: Mage.Sets/src/mage/cards/b/BewitchingLeechcraft.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return false;
+    }
+
+    @Override
+    public boolean applies(GameEvent event, Ability source, Game game) {
+        Permanent eventPermanent = game.getPermanent(event.getTargetId());
+        Permanent sourcePermanent = game.getPermanent(source.getSourceId());
+        return eventPermanent != null && sourcePermanent != null && sourcePermanent == eventPermanent;
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/b/BewitchingLeechcraft.java:107]: Not good practice to compare `Permanent` with `==` operator. Need to compare their UUIDs with `.equals()`. I think you can just use `event.getTargetId().equals(source.getSourceId()))`

**Review by xenohedron — CHANGES_REQUESTED** [review]

File: Mage.Sets/src/mage/cards/b/BewitchingLeechcraft.java (hunk 3 above)
Comment (Susucre) [reply Mage.Sets/src/mage/cards/b/BewitchingLeechcraft.java:65]: Oh right, not setting a modifier does indeed make it package private in Java.

**Review by xenohedron — APPROVED** [review]

**Pull request merged**